portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,7 +29,20 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Data page</h1>");
-    response.getWriter().println("<p>Hello Giovanni</p>");
+    // response.getWriter().println("<h1>Data page</h1>");
+    // response.getWriter().println("<p>Hello Giovanni</p>");
+
+    ArrayList<String> s = new ArrayList<String>(3);
+    s.add("One : First Element");
+    s.add("Two : Second Element");
+    s.add("Three : Third Element");
+    String json = gsonToJson(s);
+    response.getWriter().println(json);
+
+  }
+   private String gsonToJson(ArrayList<String> str) {
```

Comment:
[Style](https://google.github.io/styleguide/javaguide.html#s4.6.1-vertical-whitespace): A single blank line always appears between consecutive methods

```suggestion

  private String gsonToJson(ArrayList<String> str) {
```

Also, the indentation before `private` should be 2 spaces